docs/usage/language-constraints-upgrading.md

```diff
@@ -0,0 +1,87 @@
+
+Now also consider a library which sets its `engines` to `"node": "^18.12.0 || ^20.9.0"` because it wants to only support "LTS releases" of Node.js.
+Strictly speaking, this dependency is not compatible with the project above which has wider requirements for Node versions, and Renovate would hold back any upgrades for it.
+Should Renovate somehow reason and _assume_ that this narrower engines support is actually OK?
+What if the project _already_ used an existing version of this library "incorrectly"?
+
+A second problem is that if Renovate is unable to update the language constraints, or the user ignores or does not notice the language upgrade, then they also may not realize that many of their dependencies are out of date because Renovate is not proposing PRs.
+For example, a project may have 10 dependencies, and 8 of those have updates but all 8 require the project to update its language constraints first.
+The project administrator may think they are mostly up to date, because Renovate is not proposing PRs, but in reality 80% of their dependencies are outdated.
+
+In short, users who apply `constraintsFiltering=strict` usually understimate just how strict that will be and how many releases will be filtered out.
+
+## Transitive constraint limitations
+
+It is a common problem that a library may declare language constraints (such as the `engines` from the above examples) but then itself depend on one or more libraries which have narrower contraints (such as `"node": "^20.0.0"`).
+In such scenarios Renovate "trusts" the declaration of the library and may propose it as an update even with strict constraints filtering.
+
+For some package managers, such as npm, this incompatibility won't be detected or warned (even during lock file generation), although it may also not be a problem in your application either.
+Other package managers, such as Poetry, may detect incompatible language constraints during lock file generation and Renovate will report that as an "Artifacts update error".
+
+## Applying constraints through config
+
+Renovate supports the manual configuration of language constraints in its config. For example:
+
+```json title="Renovate config with Node.js constraints"
+{
+  "constraints": {
+    "node": "^18.0.0 || >=20.0.0"
+  }
+}
+```
+
+Defining constraints manually using Renovate config can be necessary for the following use cases:
+
+- The package manager of the project doesn't support constraints declarations, or
+- The project hassn't declared any constraints, or
+- The user wants Renovate to use different constraints to what's declared in the project.
+
+Renovate will not propose "update" PRs proposing to update any of these versions once they become outdated, so the user is responsible for keeping them updated manually.
+For this reason, configuring constraints manually in config is considered _undesirable_ and we'd prefer to fix problems in Renovate instead if that's the reason for needing it.
```

Comment:
```suggestion
For this reason, setting constraints manually in the Renovate config is _undesirable_.
We prefer to fix problems in Renovate itself, instead of you setting constraints.
```